### DISCUSSION TITLE
Add more options for launching a self-hosted web server

### DISCUSSION THREAD

**Discussion by umprex** [outline/outline#7991] [Ideas]
Now i can only use the port number, which will take up all the IPs on the server. Give the opportunity to use a socket path instead of a port, as well as an additional IP address as second argument in `server.listen()`.

**Comment by smegoff** [comment]
Can you expand on this a bit?
Outline uses nginx to reverse proxy - so you only need to use port 443 to a single IP, and use either your firewall or ufw to limit connections to 443 only. 
You wouldn't / shouldn't expose outline directly to the internet without a proxy.

**Reply by umprex** [reply]
I understand it. We will host outline on a shared hosting and nginx can proxying is only in a socket and in dedicated ip(i can’t occupy all ips). As a plus, proxying in a socket work faster.
